GlobalOptions.py

```diff
@@ -1,54 +1,46 @@
 from abc import ABC, abstractmethod
 
-
-class GlobalOptionBuilder(ABC):
-    globalOptionsList = []
-
+class GlobalOptionsBuilder(ABC):
+    '''Abstract Component Class'''
     @abstractmethod
-    def add_option(self, **kwargs):
+    def add_options(self):
         pass
 
+class DefaultGlobalOptions(GlobalOptionsBuilder):
+    '''Concrete Component Class'''
+    def __init__(self,globalOptionsList):
+        self.globalOptionsList=globalOptionsList
+    def add_options(self):
+        return self.globalOptionsList
 
-class DefaultGlobalOptions(GlobalOptionBuilder):
-
-    global_options_list: list = []
-
-    def __init__(self, global_options_list):
-        self.global_options_list = global_options_list
-
-    def add_option(self, **kwargs):
-        return self.global_options_list
+class GlobalOptionsDecorator(GlobalOptionsBuilder,ABC):
+    '''Abstract Decorator class for adding multiple global options'''
 
-
-class GlobalOptionDecorator(GlobalOptionBuilder, ABC):
-
-    _default_global_options: DefaultGlobalOptions = None
-
-    def __init__(self, default_global_options: DefaultGlobalOptions):
-        self._default_global_options = default_global_options
+    def __init__(self,globalOptionsBuilder):
```

Comment:
You need to initialize the default_global_options in the constructor. This will be the composed object which the decorator will decorate.

Reply:
Basically replace `globalOptionsBuilder` with `default_global_options`

Reply:
Or vice versa

Reply:
ha but since I am calling the add_options() within the add_options it will recursively end up decorating the default_global_options only, which in my case is the globalOptionsList only. Basically trying the below diagram.
![decorator](https://github.com/hitanshu310/PyTranscode/assets/46090325/ac351aa5-018f-490f-abfd-0114d5101111)